### PR TITLE
Bug fix : add check befor fencing

### PR DESCRIPTION
**Pull request by leo79901**
### Description

Resolve the issur : Should not shutting down a server which is already shut.
Thanks for DaanHoogland which provide code.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?
I am so sorry about this. I don't know much about Java, thus I have not tested...

### CODE REVIEW

File: plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHAProvider.java [hunk 1]
```diff
@@ -84,18 +84,24 @@ public boolean recover(Host r) throws HARecoveryException {
...
+                if (Host.Status.DOWN.equals(r.getStatus())) {
+                    LOG.info("Host " + r.getName() + " is already down. Returning success.");
+                    return true;
+                } else {
+                    final OutOfBandManagementResponse resp = outOfBandManagementService.executePowerOperation(r, PowerOperation.OFF, null);
+                    return resp.getSuccess();
+                }
             } else {
-                logger.warn("OOBM fence operation failed for this host " + r.getName());
+                LOG.warn("OOBM fence operation failed for this host " + r.getName());
```
Comment (weizhouapache) [inline plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHAProvider.java:99]: @leo79901 
should it be `logger` instead of `LOG` ?

Comment (leo79901) [reply plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHAProvider.java]: Yes , you are right.

File: plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHAProvider.java [hunk 2]
```diff
@@ -84,11 +85,18 @@ public boolean recover(Host r) throws HARecoveryException {
 
     @Override
     public boolean fence(Host r) throws HAFenceException {
-
+
```
Comment (DaanHoogland) [inline plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHAProvider.java:88]:
```suggestion

```

**Comment by DaanHoogland** [comment]
sorry to bug you with this, the linter says no. (just some trailing whitespace.)

**Comment by DaanHoogland** [comment]
@slavkap @weizhouapache any sugestions for testing?

Also, I marked this for 4.21 @leo79901 , but it seems to me it can go onto 4.19. I think you'd like this fix sooner rather than later, right?

if you work with git regularly this is an easy thing to do, but as you say you are not a developer:
```
git rebase --onto 4.19 4ac4d9cf29abd83526093d162424da0419496297 3a6904094da9f84264c5a80659d334a561ff92b2
```
but these will change when you add work to it. Please let us know if you need help with this.

**Comment by leo79901** [comment]
@DaanHoogland 
Thanks for your help! 
Yes, I really want to use this in our environment which is 4.19.1.2. But i found that 'log' in 4.19 is 'LOG', and in 4.20 it is 'logger'. Do I need to edit the code again?

**Comment by DaanHoogland** [comment]
> @DaanHoogland Thanks for your help! Yes, I really want to use this in our environment which is 4.19.1.2. But i found that 'log' in 4.19 is 'LOG', and in 4.20 it is 'logger'. Do I need to edit the code again?

ai, yes. There has been a big logging change between 4.19 and 4.20 and you would indeed need to edit it. If you are planning to upgrade anyway and can life without till then you can change the commands to 
```
git rebase --onto 4.20 4ac4d9cf29abd83526093d162424da0419496297 67e33f60146c923682d82c184c3fcd47cfd039c1
```
(changed to include the latest white space fix)

**Comment by leo79901** [comment]
Thanks really for your help.
I have run the command.
It's my great honor to contribute to the project. I will do more within my capacity in the future.

**Comment by DaanHoogland** [comment]
> Thanks really for your help. I have run the command. It's my great honor to contribute to the project. I will do more within my capacity in the future.

great @leo79901 , next step is to force push to your fork (git push --force)
and to edit this PR to point to the older branch (i can do that to show you)

In the future it will be better to create a branch locally and create PRs from such a branch. You are now having your changes on your local branch called `main`. This is not wroing but might be confusing.

**Comment by leo79901** [comment]
I'm terribly sorry for bothering you several times. I think I need to learn more about Git before submitting the PR. Could you please help me complete this operation this time?

**Comment by DaanHoogland** [comment]
> I'm terribly sorry for bothering you several times. I think I need to learn more about Git before submitting the PR. Could you please help me complete this operation this time?

no problem @leo79901 . I saw your other PR as well. I will copy your branch and create a PR from your code.

**Comment by DaanHoogland** [comment]
@leo79901 I created #10059 for you, and will close this one. cc @weizhouapache

**Pull request closed without merging**